src/internals/shape.point.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2017 NAVER Corp.
+ * billboard.js project is licensed under the MIT license
+ */
+import ChartInternal from "./ChartInternal";
+import {isFunction, extend} from "./util";
+
+const POINT_TYPES = ["circle", "rectangle"];
+
+extend(ChartInternal.prototype, {
+	hasValidPointType() {
+		return POINT_TYPES.includes(this.config.point_type);
+	},
+
+	hasValidPointDrawMethods() {
+		return this.config.point_type === "custom" &&
+			isFunction(this.config.point_create) && isFunction(this.config.point_update);
+	},
+
+	circle: {
+		create(element, cssClassFn, sizeFn, fillStyleFn) {
+			return element.enter().append("circle")
+				.attr("class", cssClassFn)
+				.attr("r", sizeFn)
+				.style("fill", fillStyleFn);
+		},
+
+		update(element, xPosFn, yPosFn, opacityStyleFn, fillStyleFn,
+			withTransition, flow, selectedCircles) {
+			let mainCircles;
+
+			if (withTransition) {
+				const transitionName = Math.random().toString();
+
+				if (flow) {
+					mainCircles = element
+						.attr("cx", xPosFn)
+						.transition(transitionName)
+						.attr("cx", xPosFn)
+						.attr("cy", yPosFn)
+						.transition(transitionName)
+						.style("opacity", opacityStyleFn)
+						.style("fill", fillStyleFn);
+				} else {
+					mainCircles = element
+						.transition(transitionName)
+						.attr("cx", xPosFn)
+						.attr("cy", yPosFn)
+						.transition(transitionName)
+						.style("opacity", opacityStyleFn)
+						.style("fill", fillStyleFn);
+				}
+
+				selectedCircles.transition(Math.random().toString());
+			} else {
+				mainCircles = element
+					.attr("cx", xPosFn)
+					.attr("cy", yPosFn)
+					.style("opacity", opacityStyleFn)
+					.style("fill", fillStyleFn);
+			}
+
+			return mainCircles;
+		}
+	},
+
+	rectangle: {
+		create(element, cssClassFn, sizeFn, fillStyleFn) {
+			const rectSizeFn = d => sizeFn(d) * 2.0;
```

Comment:
It's using `config.ponint_r` value to determine the size of the data point.
But, when do mouseover(or interaction) on data point, isn't get changed bigger than normal size.

Is because of below code. Is changing `r` attribute value, which only works for `<circle>`
https://github.com/naver/billboard.js/blob/cf49d71c68a040694ff7f7d458ea6c9a65da1e98/src/internals/shape.line.js#L526-L537

I know that is not easy to handle that with squares and customs, but I think it should have a compatibility with the others `point` options.

Reply:
@netil thant's a good point.
That said I have no idea how this could be applied to custom shapes (for example `polygon`),  I did make an update to the current code so that it would only update the `r` attribute for `circle` and not other "shapes" while we think about a solution. (Maybe using css `transform` and `scale` is an option?

Reply:
That's good idea. I'll dig into `transform:scale` if this works :)

Reply:
With `transform-origin` and `transform:scale()` can make it happen.
But, in the usage of `r` option, need to be discussed more.

The actual `r` means radius, so for other types it means `r*2`.
I think this can cause a confusion and the real meaning also doesn't fits.

The alternative that I though was:
- use `r` value in different way when isn't circle.
   Instead of radius, it treats as 'scale'
- set another option key like 'size'
   But, still there'll be a confusion between 'r' and 'size'

```js
expandCircles(i, id, reset) {
	...

	if ($$.config.point_type === "circle") {
		circles.attr("r", r);
	} else {
		let scale = r(circles);

		if (!isUndefined($$.config.point_focus_expand_r)) {
			scale /= 2;
		}

		scale *= 100 / $$.config.point_r / 100;

		circles
			.style("transform-origin", "center")
			.style("transform", `scale(${scale})`);
	}
},

unexpandCircles(i) {
	...

	if ($$.config.point_type === "circle") {
		circles.attr("r", r);
	} else {
		circles
			.style("transform-origin", null)
			.style("transform", null);
	}
},
```

Reply:
@netil from reading the code, and seeing it was rendering svg "circle" elements, it was quite clear to me that `r` meant `radius`. I like the approach of using scale for other shapes rather than having a `size` option, but that's just my opinion ;)